### DISCUSSION TITLE
how can I check permission is used or not before delete?

### DISCUSSION THREAD

**Discussion by Logical-Developer** [spatie/laravel-permission#1831] [Q&A]
Hi
I want delete one specific permission, how can I check permission is used or not?
I know it should be done in 3 parts:
1. Roles Table
2. role_has_permissions Table
3. finally check user with this Permission

in first step I checked roles have this permission:
$permission->roles()->count()

Please help me how can I **check** before **delete** permission ?

**Comment by drbyte** [comment]
`$permission->roles->count();` // number of roles using that permission
`$permission->users->count();` // number of users with that direct permission
